File: BeagleBone/Black/gpiod/toggle1.py
#!/usr/bin/env python3
# //////////////////////////////////////
# 	toggle.c
#  Toggles the P9_14 pin as fast as it can. P9_14 is line 18 on chip 1.
# 	Wiring:	Attach an oscilloscope to P9_14 to see the squarewave or 
#          uncomment the usleep and attach an LED.
# 	Setup:	sudo apt uupdate; pip3 install gpiod
# 	See:	https://pypi.org/project/gpiod/0.5.4/
# //////////////////////////////////////

import gpiod
import sys
import time

if len(sys.argv) > 2:
    LED_CHIP = sys.argv[1]
    LED_LINE_OFFSET = int(sys.argv[2])
else:
    print('''Usage:\npython3 blink.py <chip> <line offset>''')
    LED_CHIP = 1
    LED_LINE_OFFSET = 18  # P9_14
    # sys.exit()

chip = gpiod.chip(LED_CHIP)
led = chip.get_line(LED_LINE_OFFSET)

config = gpiod.line_request()
config.consumer = "Blink"
config.request_type = gpiod.line_request.DIRECTION_OUTPUT

led.request(config)

while True:
    led.set_value(0)
    # time.sleep(0.1)
    led.set_value(1)
    # time.sleep(0.1)

File: BeagleBone/Black/gpiod/toggle1.c
// //////////////////////////////////////
// 	toggle1.c
//  Toggles the P9_14 pin as fast as it can. P9_14 is line 18 on chip 1.
// 	Wiring:	Attach an oscilloscope to P9_14 to see the squarewave or 
//          uncomment the usleep and attach an LED.
// 	Setup:	sudo apt uupdate; sudo apt install libgpiod-dev
// 	See:	https://github.com/starnight/libgpiod-example/blob/master/libgpiod-led/main.c
// //////////////////////////////////////
#include <gpiod.h>
#include <stdio.h>
#include <unistd.h>

#define	CONSUMER	"Consumer"

int main(int argc, char **argv)
{
	int chipnumber = 1;
	unsigned int line_num = 18;	// GPIO Pin P9_14
	unsigned int val;
	struct gpiod_chip *chip;
	struct gpiod_line *line;
	int i, ret;

	chip = gpiod_chip_open_by_number(chipnumber);
	if (!chip) {
		perror("Open chip failed\n");
		goto end;
	}

	line = gpiod_chip_get_line(chip, line_num);
	if (!line) {
		perror("Get line failed\n");
		goto close_chip;
	}

	ret = gpiod_line_request_output(line, CONSUMER, 0);
	if (ret < 0) {
		perror("Request line as output failed\n");
		goto release_line;
	}

	/* Blink */
	val = 0;
	while(1) {
		ret = gpiod_line_set_value(line, val);
		if (ret < 0) {
			perror("Set line output failed\n");
			goto release_line;
		}
		// printf("Output %u on line #%u\n", val, line_num);
		// usleep(100000);		// Number of microseconds to sleep
		val = !val;
	}

release_line:
	gpiod_line_release(line);
close_chip:
	gpiod_chip_close(chip);
end:
	return 0;
}
